### PR TITLE
Add function to deploy contracts for development

### PR DESCRIPTION
**Pull request by geigerzaehler**
* Add a function `deployDev` that deploys development contract infrastructure.
* Use `deployDev` in tests
* Expose `deployDev` as part of the package
* Add ENS helper functions to better deal with name and label hashes
* Add package tarball to build artifacts so we can use it from other projects (i.e. upstream)

### CODE REVIEW

**Review by cloudhead — APPROVED** [review]
Nice!

File: src/deploy.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  const router = await new DummyRouterFactory(signer).deploy(rad.address);
+
+  const exchange = await new ExchangeFactory(signer).deploy(
+    rad.address,
+    router.address,
+    oracle.address
+  );
+
+  const registrar = await new RegistrarFactory(signer).deploy(
```
Comment (NunoAlexandre) [inline src/deploy.ts:45]:
```suggestion
  const signerAddr = await signer.getAddress();
  const oracle = await new DummyPriceOracleFactory(signer).deploy(1);
  const ens = await new DummyEnsRegistryFactory(signer).deploy();
  const rad = await new RadFactory(signer).deploy(signerAddr, 1e6);
  const router = await new DummyRouterFactory(signer).deploy(rad.address);
  const exchange = await new ExchangeFactory(signer).deploy(
    rad.address,
    router.address,
    oracle.address
  );
  const registrar = await new RegistrarFactory(signer).deploy(
```

File: src/deploy.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  await submitOk(
+    ens.setSubnodeOwner(
+      ensUtils.nameHash(""),
+      ensUtils.labelHash("eth"),
+      signerAddr
+    )
+  );
+
+  await submitOk(
```
Comment (NunoAlexandre) [inline src/deploy.ts:70]:
```suggestion

  await registrar.deployed();
  await oracle.deployed();
  await ens.deployed();
  await rad.deployed();
  await router.deployed();
  await exchange.deployed();
  await (await rad.connect(signer).transfer(router.address, 1e3)).wait();
  await submitOk(
    ens.setSubnodeOwner(
      ensUtils.nameHash(""),
      ensUtils.labelHash("eth"),
      signerAddr
    )
  );
  await submitOk(
```

File: test/registrar.test.ts [hunk 3]
```diff
@@ -1,95 +1,27 @@
...
-    );
-
-    await oracle.deployed();
-    await ens.deployed();
-    await registrar.deployed();
-    await rad.deployed();
-
-    const ethNode = await registrar.namehash(zeroNode, ethLabel);
-    const cloudheadNode = await registrar.namehash(radicleNode, cloudheadLabel);
+    const {rad, registrar, ens} = await deployDev(owner);
```
Comment (NunoAlexandre) [inline test/registrar.test.ts:10]: neat :ok_hand:

**Review by NunoAlexandre — APPROVED** [review]
Looks good from what I can tell :ok_hand: 

Just left a few nit suggestions, feel free to skip them, they are subjective organizational suggestions.

File: src/deploy.ts (hunk 1 above)
Comment (cloudhead) [reply src/deploy.ts:45]: Was gonna suggest that :+1:

**Pull request merged**